stdlib/importlib/abc.pyi

```diff
@@ -82,24 +91,82 @@ if sys.version_info >= (3, 7):
         def contents(self) -> Iterator[str]: ...
 
 if sys.version_info >= (3, 9):
+    _P = TypeVar("_P", bound=Traversable)
     @runtime_checkable
     class Traversable(Protocol):
         @abstractmethod
-        def iterdir(self) -> Iterator[Traversable]: ...
+        def is_dir(self) -> bool: ...
         @abstractmethod
-        def read_bytes(self) -> bytes: ...
+        def is_file(self) -> bool: ...
         @abstractmethod
-        def read_text(self, encoding: Optional[str] = ...) -> str: ...
+        def iterdir(self: _P) -> Generator[_P, None, None]: ...
```

Comment:
The old `Iterator[Traversable]` was arguably better, as it is more permissive. If someone has their own path class whose `iterdir()` returns an iterator that isn't a generator, we should still accept it as a `Traversible`.

Reply:
Changed